### PR TITLE
feat(class_metadata): inline templateUrl/styleUrls + default-on, matching ngc

### PR DESCRIPTION
**Pull request by ashley-hunter**
## Summary

Brings OXC's `setClassMetadata` emission into byte-exact parity with Angular's `ngc`, so TestBed JIT recompilation works without users having to discover and flip a flag.

Three threads of work:

1. **Inline `templateUrl` / `styleUrls` / `styleUrl` into the metadata args** (`d1d3ac8`). Pre-existing emission preserved the raw resource literals, so Angular's `componentNeedsResolution(metadata)` check tripped at TestBed time (`'Component is not resolved: templateUrl…'`).

2. **Default `emit_class_metadata` to `true` at all three layers** (Rust crate, NAPI binding, Vite plugin) — `ngc` always emits the call wrapped in `(typeof ngDevMode === "undefined" || ngDevMode) && …`, so production bundles tree-shake it. Off-by-default silently broke TestBed/Vitest setups.

3. **Match `ngc`'s exact semantics**, found via deeper audit:
   - `Map.delete('templateUrl')` + `Map.set('template', …)` ordering (append at end, not in-place) — the e2e compare does string-equality of the decorators payload so the in-place divergence would have silently miscompared.
   - Deduplicate when source illegally contained both `template` and `templateUrl`.
   - Gate inlining on the decorator's name being `Component` (matches Angular's `if (dec.name !== 'Component') return dec;`) — without this, `@Inject({ templateUrl: '…' })` had its keys silently stripped.
   - `resolve_template` precedence: `templateUrl` wins over inline `template` (matches `parseTemplateDeclaration`); OXC was preferring inline, which is JIT's behavior, not AOT.

The Rust port now reads as a near-line-by-line mirror of `transformDecoratorResources`, with citations to the upstream source in the doc comments.

## Empirical ngc parity

All 686 non-skipped fixtures across 22 categories of the e2e ngc-compare suite match byte-for-byte: animations, bindings, class-metadata, control-flow, defer, edge-cases, host-bindings, host-directives, i18n, pipes, providers, regressions, schemas, styles, templates, etc.

## Test plan

- [x] `cargo check --all-features`
- [x] `cargo test` (21 binaries, 2685+ tests, all ok)
- [x] `cargo fmt --all -- --check`
- [x] `cargo run -p oxc_angular_conformance && git diff --exit-code` (1252/1252, git-clean)
- [x] `pnpm build-dev` + `pnpm --filter ./napi/angular-compiler build:ts`
- [x] `pnpm test` (193/193 vitest)
- [x] `pnpm check` (oxfmt + oxlint, clean)
- [x] `pnpm test:e2e` (34/34 playwright)
- [x] `pnpm --filter @oxc-angular/compare compare --fixtures` (686/686 against ngc 21.2.14)

## Notes for reviewers

- Behavior change: every component now emits `setClassMetadata` by default. Production strips it via `ngDevMode`. Anyone who actively wants the old behavior passes `emitClassMetadata: false`.
- 14 integration snapshots updated to include the new (correctly-guarded) `setClassMetadata` block.
- One assertion narrowed (`test_standalone_component_omits_standalone_field`) to scope its `contains("standalone:true")` check to the `ɵɵdefineComponent` literal — the metadata block faithfully preserves the source `standalone: true`, which is what ngc does.
- Known limitations locked in by tests: spread elements containing resource fields (`@Component({ ...config, … })`) pass through unchanged; fixing requires structural pre-extraction spread resolution upstream of metadata.

### CODE REVIEW

**Comment by Brooooooklyn** [comment]
@codex review

File: crates/oxc_angular_compiler/src/component/transform.rs [hunk 1]
```diff
@@ -2021,6 +2024,8 @@ pub fn transform_angular_file(
                                             allocator,
                                             &[decorator],
                                             Some(source),
+                                            Some(template),
+                                            Some(metadata.styles.as_slice()),
```
[the remark this answers is not in the document]
Comment (ashley-hunter) [reply crates/oxc_angular_compiler/src/component/transform.rs:2028]: Thanks - the mechanism you identified is accurate: `compile_component_full` appends template `<style>` content to `metadata.styles`, and this call site passes that post-template slice into `setClassMetadata`.

However, the premise that this diverges from ngc isn't correct - **ngc includes template `<style>` content in `setClassMetadata` too.** Verified against the installed `@angular/compiler-cli@21.2.14` (`ComponentDecoratorHandler`):

```js
styles.push(resourceStr);          // resolved styleUrls
styles.push(...inlineStyles);      // decorator inline styles
styles.push(...litStyles);
if (template.styles.length > 0) {
  styles.push(...template.styles); // template <style> tags → same array
}
// ...the same `styles` array is then handed to class metadata:
transformDecoratorResources(dec, component, styles, template)
```

The single `styles` array - with `template.styles` already pushed in - feeds both the compiled `ɵcmp` and the class-metadata transform. So ngc's `ɵsetClassMetadata` carries `styleUrls` + inline `styles` + template `<style>` content, in that order.

Since the goal here is parity with ngc, the current behavior is correct - restricting class metadata to pre-template styles would *introduce* a divergence rather than remove one.

**Review by Brooooooklyn — APPROVED** [review]
